opensearch_py_ml/sentence_transformer_model/sentencetransformermodel.py

```diff
@@ -0,0 +1,677 @@
+                num_training_steps=total_steps,
+            )
+
+            loss = []
+            init_time = time.time()
+
+            print("Start training without accelerator...")
+            print(f"The number of training epoch are {num_epochs}")
+            print(f"The total number of steps training epoch are {steps_size}")
+
+            for epoch in range(num_epochs):
+                random.shuffle(train_examples)
+                print("Training epoch " + str(epoch) + "...")
+                for j in tqdm(range(steps_size), total=steps_size):
+                    batch = []
+                    batch_q = []
+                    for example in train_examples[
+                        j * batch_size : (j + 1) * batch_size
+                    ]:
+                        batch_q.append(example.texts[1])
+                        batch.append(example.texts[0])
+
+                    out_q = model.tokenize(batch_q)
+                    for key in out_q.keys():
+                        out_q[key] = out_q[key].to(device)
+                    Y = model(out_q)["sentence_embedding"]
+
+                    out = model.tokenize(batch)
+                    for key in out.keys():
+                        out[key] = out[key].to(device)
+                    X = model(out)["sentence_embedding"]
+
+                    XY = torch.exp(torch.matmul(X, Y.T) / (X.shape[1]) ** 0.5)
+                    num = torch.diagonal(XY)
+
+                    den0 = torch.sum(XY, dim=0)
+                    den1 = torch.sum(XY, dim=1)
+
+                    train_loss = -torch.sum(torch.log(num / den0)) - torch.sum(
+                        torch.log(num / den1)
+                    )
+                    loss.append(train_loss.item())
+                    train_loss.backward()
+                    optimizer.step()
+                    scheduler.step()
+                    optimizer.zero_grad()
+
+                    if not j % 500 and j != 0:  # MS: change 500
+                        plt.plot(loss[::100])
+                        plt.show()
+
+        # saving the pytorch model and the tokenizers.json file is saving at this step
+        model.save("trained_pytorch_model/")
+        device = "cpu"
+        cpu_model = model.to(device)
+        print(f"Total training time: {time.time() - init_time}")
+
+        for key in out_q.keys():
+            out_q[key] = out_q[key].to(device)
+
+        traced_cpu = torch.jit.trace(
+            cpu_model,
+            (
+                {
+                    "input_ids": out_q["input_ids"],
+                    "attention_mask": out_q["attention_mask"],
+                }
+            ),
+            strict=False,
+        )
+
+        print("Preparing model to save")
+        torch.jit.save(traced_cpu, output_model_path)
+
+        print("Model saved to path: " + output_model_path)
+        return model
+
+    def zip_model(self, model_path: str = None, zip_file_name: str = None) -> None:
```

Comment:
How about add a method to export huggingface sentence transformer model to local and create model zip? So the ml-commons python client can call this method , then upload the zip directly.

Reply:
added function save_as_pt() , 
if pass the model in the parameters, it will convert as pt format, download the pt model, save its tokenizers.json file, and zip file as output.
if no model is passed, it download the model from hugging face and do the same as above.

Reply:
resolved in 
[added save_as_pt function](https://github.com/opensearch-project/opensearch-py-ml/pull/18/commits/094339cca0a561d3ecd4edf74ce803b77e973239)